File: BackEnd/backend/src/main/java/com/fitapp/backend/auth/aplication/dto/response/UserResponse.java
package com.fitapp.backend.auth.aplication.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.LocalDateTime;

import com.fitapp.backend.infrastructure.persistence.entity.enums.Role;
import com.fitapp.backend.suscription.aplication.dto.response.SubscriptionResponse;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class UserResponse {
    @JsonProperty("id")
    private Long id;
    
    @JsonProperty("createdAt")
    private LocalDateTime createdAt;
    
    @JsonProperty("updatedAt")
    private LocalDateTime updatedAt;
    
    @JsonProperty("email")
    private String email;
    
    @JsonProperty("fullName")
    private String fullName;
    
    @JsonProperty("role")
    private Role role;
    
    @JsonProperty("isActive")
    private boolean isActive;
    
    @JsonProperty("subscription")
    private SubscriptionResponse subscription;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/user/request/UserResponse.kt
package com.fitapp.appfit.response.user.request

import com.fitapp.appfit.response.subscription.response.SubscriptionResponse
import com.google.gson.annotations.SerializedName

data class UserResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("email") val email: String,
    @SerializedName("fullName") val fullName: String?,
    @SerializedName("role") val role: String,
    @SerializedName("isActive") val isActive: Boolean,
    @SerializedName("emailVerified") val emailVerified: Boolean = false,
    @SerializedName("subscription") val subscription: SubscriptionResponse?
)